lumicks/pylake/channel.py

```diff
@@ -504,6 +504,137 @@ def plot(self, start=None, **kwargs):
+            String to annotate with
+        annotation_direction : "horizontal" | "vertical" | None
+            Which direction to show the text when annotating.
+        color : matplotlib.color | None
+            Matplotlib color
+        label : str | None
+            Label to use in legend entry if a legend is plotted
+        start : int | None
+            Plot with respect to a different "zero" timestamp.
+        alpha : float
+            Opacity (0 is transparent, 1.0 is opaque). Default: 0.5.
+        **kwargs :
+            Arguments forwarded to `matplotlib.axvline` and `matplotlib.axvspan`.
+
+        Examples
+        --------
+        ::
+
+            from lumicks import pylake
+
+            f = lk.File("data.h5")
+
+            plt.figure()
+            slc = ["Distance"]["Distance 1"]
+            slc.plot()
+
+            # Draws the time region spanned by marker 1
+            slc.highlight_time_range(f.markers["1"])
+
+            # Draws an annotation at the 1-second mark
+            slc.highlight_time_range("1s", annotation="first second")
+
+            # Draws a highlighted region from 1 to 3 seconds and marks it with the text "region"
+            slc.highlight_time_range(("1s", "3s"), annotation="region")
+
+            # Draws a highlighted region from 4 to 5 seconds, marks it, but forces text vertically
+            slc.highlight_time_range(("1s", "3s"), annotation="region", annotation_direction="vertical")
+
+
+            plt.figure()
+            f.force1x.plot()
+
+            item = f.force1x.calibration[1]
+
+            # Fills the region between calibration_item.start and calibration_item.stop
+            f.force1x.highlight_time_range(item, annotation="calibration 1")
+
+            # Draws a line with annotation at the time the calibration was applied.
+            f.force1x.highlight_time_range(item.applied_at, annotation="calibration 1 applied")
+        """
+        import matplotlib.pyplot as plt
+
+        def to_seconds(timestamp):
+            start_ts = self.start if start is None else start
+
+            if isinstance(timestamp, str):
+                timestamp = to_timestamp(timestamp, start_ts, self.stop)
+
+            return (timestamp - start_ts) / 1e9
+
+        def get_time_range(time_item):
+            if hasattr(time_item, "start") and hasattr(time_item, "stop"):
+                return to_seconds(time_item.start), to_seconds(time_item.stop)
+            elif np.isscalar(time_item) == 1:
+                return to_seconds(time_item), None
+            elif len(time_item) == 2:
+                return to_seconds(time_item[0]), to_seconds(time_item[1])
+            else:
+                raise ValueError(
+                    "Provided item must be either timestamp, two timestamps or an item with a "
+                    f"start and stop property, got {time_item} instead."
+                )
+
+        start_time, stop_time = get_time_range(item)
+        if annotation:
```

Comment:
I'd split out the annotation handling into a separate function, esp. because it's optional.

Reply:
Yeah, that's fair. I'll split it to its own function. It does indeed help readability.

I do think it should be part of this function in terms of public API though.

Right now, I don't think splitting this over two API calls helps discoverability and the behavior is pretty linked to this range annotation (with all the logic regaring start and optional stop).